source/boundary_traction/interface.cc

```diff
@@ -78,20 +61,28 @@ namespace aspect
                                      const Point<dim> &position,
                                      const Tensor<1,dim> &normal_vector) const
     {
-      typename std::map<types::boundary_id,std::vector<std::unique_ptr<BoundaryTraction::Interface<dim>>>>::const_iterator boundary_plugins =
-        boundary_traction_objects.find(boundary_indicator);
+      Tensor<1,dim> traction;
 
-      Assert(boundary_plugins != boundary_traction_objects.end(),
+      [[maybe_unused]] bool found_plugin = false;
+      unsigned int i=0;
+      for (const auto &plugin: this->plugin_objects)
+        {
+          if (this->boundary_indicators[i])
+            {
+              const Tensor<1,dim> plugin_traction = plugin->boundary_traction(boundary_indicator,
+                                                                              position,
+                                                                              normal_vector);
+              for (unsigned int d=0; d<dim; ++d)
+                if (this->component_masks[i][d])
+                  traction[d] += plugin_traction[d];
+            }
+        }
```

Comment:
Something is not right with this code. I don't see you incrementing `i` anywhere. It's also not clear that `this-boundary_indicators[i]` is a `bool`. Are you forgetting to compare it against `boundary_indicator`?